### PR TITLE
flavours: add missing libiconv dependency on darwin

### PR DESCRIPTION
**Pull request by moni-dz**
###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/misc/flavours/default.nix [hunk 1]
```diff
@@ -18,6 +18,7 @@ rustPlatform.buildRustPackage rec {
     homepage = "https://github.com/Misterio77/flavours";
     changelog = "https://github.com/Misterio77/flavours/releases/tag/v${version}";
     license = licenses.mit;
+    broken = stdenv.hostPlatform.isDarwin;
```
Comment (jonringer) [inline pkgs/applications/misc/flavours/default.nix:21]: do you mind adding a comment?

Comment (fortuneteller2k) [reply pkgs/applications/misc/flavours/default.nix]: sure

**Comment by jonringer** [comment]
For the comment, I meant more as to the nature as to why darwin builds should be marked as failing

**Comment by fortuneteller2k** [comment]
> For the comment, I meant more as to the nature as to why darwin builds should be marked as failing, or just mark `platforms` as linux

Just found out that it fails due to missing `libiconv` 

> For the comment, I meant more as to the nature as to why darwin builds should be marked as failing, or just mark `platforms` as linux

Looking at the build log, it says `libiconv` is missing on darwin, so I decided to add it as a dependency instead of marking it broken.

```
Compiling palette v0.5.0
error: linking with `/nix/store/l3zvd3g9jwc7vq4jnfnz7xh7zvvplxn7-clang-wrapper-7.1.0/bin/cc` failed: exit code: 1
  |
  = note: "/nix/store/l3zvd3g9jwc7vq4jnfnz7xh7zvvplxn7-clang-wrapper-7.1.0/bin/cc" "-m64" "-arch" "x86_64" "-L" "/nix/store/j21vazwy4w6c20mhlqpafyxbwj3q226j-rustc-1.50.0/lib/rustlib/x86_64-apple-darwin/lib" "/private/tmp/nix-build-flavours-0.4.0.drv-0/source/target/release/build/palette-a97e528154b37823/build_script_main-a97e528154b37823.build_script_main.c6zlq865-cgu.0.rcgu.o" "/private/tmp/nix-build-flavours-0.4.0.drv-0/source/target/release/build/palette-a97e528154b37823/build_script_main-a97e528154b37823.build_script_main.c6zlq865-cgu.1.rcgu.o" "/private/tmp/nix-build-flavours-0.4.0.drv-0/source/target/release/build/palette-a97e528154b37823/build_script_main-a97e528154b37823.build_script_main.c6zlq865-cgu.10.rcgu.o" "/private/tmp/nix-build-flavours-0.4.0.drv-0/source/target/release/build/palette-a97e528154b37823/build_script_main-a97e528154b37823.build_script_main.c6zlq865-cgu.11.rcgu.o" "/private/tmp/nix-build-flavours-0.4.0.drv-0/source/target/release/build/palette-a97e528154b37823/build_script_main-a97e528154b37823.build_script_main.c6zlq865-cgu.12.rcgu.o" "/private/tmp/nix-build-flavours-0.4.0.drv-0/source/target/release/build/palette-a97e528154b37823/build_script_main-a97e528154b37823.build_script_main.c6zlq865-cgu.13.rcgu.o" "/private/tmp/nix-build-flavours-0.4.0.drv-0/source/target/release/build/palette-a97e528154b37823/build_script_main-a97e528154b37823.build_script_main.c6zlq865-cgu.14.rcgu.o" "/private/tmp/nix-build-flavours-0.4.0.drv-0/source/target/release/build/palette-a97e528154b37823/build_script_main-a97e528154b37823.build_script_main.c6zlq865-cgu.15.rcgu.o" "/private/tmp/nix-build-flavours-0.4.0.drv-0/source/target/release/build/palette-a97e528154b37823/build_script_main-a97e528154b37823.build_script_main.c6zlq865-cgu.2.rcgu.o" "/private/tmp/nix-build-flavours-0.4.0.drv-0/source/target/release/build/palette-a97e528154b37823/build_script_main-a97e528154b37823.build_script_main.c6zlq865-cgu.3.rcgu.o" "/private/tmp/nix-build-flavours-0.4.0.drv-0/source/target/release/build/palette-a97e528154b37823/build_script_main-a97e528154b37823.build_script_main.c6zlq865-cgu.4.rcgu.o" "/private/tmp/nix-build-flavours-0.4.0.drv-0/source/target/release/build/palette-a97e528154b37823/build_script_main-a97e528154b37823.build_script_main.c6zlq865-cgu.5.rcgu.o" "/private/tmp/nix-build-flavours-0.4.0.drv-0/source/target/release/build/palette-a97e528154b37823/build_script_main-a97e528154b37823.build_script_main.c6zlq865-cgu.6.rcgu.o" "/private/tmp/nix-build-flavours-0.4.0.drv-0/source/target/release/build/palette-a97e528154b37823/build_script_main-a97e528154b37823.build_script_main.c6zlq865-cgu.7.rcgu.o" "/private/tmp/nix-build-flavours-0.4.0.drv-0/source/target/release/build/palette-a97e528154b37823/build_script_main-a97e528154b37823.build_script_main.c6zlq865-cgu.8.rcgu.o" "/private/tmp/nix-build-flavours-0.4.0.drv-0/source/target/release/build/palette-a97e528154b37823/build_script_main-a97e528154b37823.build_script_main.c6zlq865-cgu.9.rcgu.o" "-o" "/private/tmp/nix-build-flavours-0.4.0.drv-0/source/target/release/build/palette-a97e528154b37823/build_script_main-a97e528154b37823" "/private/tmp/nix-build-flavours-0.4.0.drv-0/source/target/release/build/palette-a97e528154b37823/build_script_main-a97e528154b37823.1j4d2g8jgsdgejeb.rcgu.o" "-Wl,-dead_strip" "-nodefaultlibs" "-L" "/private/tmp/nix-build-flavours-0.4.0.drv-0/source/target/release/deps" "-L" "/nix/store/j21vazwy4w6c20mhlqpafyxbwj3q226j-rustc-1.50.0/lib/rustlib/x86_64-apple-darwin/lib" "/private/tmp/nix-build-flavours-0.4.0.drv-0/source/target/release/deps/libphf_codegen-bec407103c258ae6.rlib" "/private/tmp/nix-build-flavours-0.4.0.drv-0/source/target/release/deps/libphf_generator-f6a3c54fadeea6f9.rlib" "/private/tmp/nix-build-flavours-0.4.0.drv-0/source/target/release/deps/librand-65a2780c1111623e.rlib" "/private/tmp/nix-build-flavours-0.4.0.drv-0/source/target/release/deps/librand_pcg-fd403836cfe17872.rlib" "/private/tmp/nix-build-flavours-0.4.0.drv-0/source/target/release/deps/librand_chacha-7d166984155c45ce.rlib" "/private/tmp/nix-build-flavours-0.4.0.drv-0/source/target/release/deps/libppv_lite86-8ed6260b62ada567.rlib" "/private/tmp/nix-build-flavours-0.4.0.drv-0/source/target/release/deps/librand_core-d5833829bb93f25b.rlib" "/private/tmp/nix-build-flavours-0.4.0.drv-0/source/target/release/deps/libgetrandom-ba8381dedb87af1c.rlib" "/private/tmp/nix-build-flavours-0.4.0.drv-0/source/target/release/deps/liblibc-54839bbfd187cdcf.rlib" "/private/tmp/nix-build-flavours-0.4.0.drv-0/source/target/release/deps/libcfg_if-24d19c64550d3577.rlib" "/private/tmp/nix-build-flavours-0.4.0.drv-0/source/target/release/deps/libphf_shared-1c77688f73fdb0be.rlib" "/private/tmp/nix-build-flavours-0.4.0.drv-0/source/target/release/deps/libsiphasher-9f87752a68ab3b30.rlib" "/nix/store/j21vazwy4w6c20mhlqpafyxbwj3q226j-rustc-1.50.0/lib/rustlib/x86_64-apple-darwin/lib/libstd-1356eb4a7f983e16.rlib" "/nix/store/j21vazwy4w6c20mhlqpafyxbwj3q226j-rustc-1.50.0/lib/rustlib/x86_64-apple-darwin/lib/libpanic_unwind-8ae047be503d7619.rlib" "/nix/store/j21vazwy4w6c20mhlqpafyxbwj3q226j-rustc-1.50.0/lib/rustlib/x86_64-apple-darwin/lib/libobject-ab8fafc452749532.rlib" "/nix/store/j21vazwy4w6c20mhlqpafyxbwj3q226j-rustc-1.50.0/lib/rustlib/x86_64-apple-darwin/lib/libaddr2line-f75e34ad235373a3.rlib" "/nix/store/j21vazwy4w6c20mhlqpafyxbwj3q226j-rustc-1.50.0/lib/rustlib/x86_64-apple-darwin/lib/libgimli-ed4b8c38c97ddcd6.rlib" "/nix/store/j21vazwy4w6c20mhlqpafyxbwj3q226j-rustc-1.50.0/lib/rustlib/x86_64-apple-darwin/lib/librustc_demangle-9ca625a2899d3719.rlib" "/nix/store/j21vazwy4w6c20mhlqpafyxbwj3q226j-rustc-1.50.0/lib/rustlib/x86_64-apple-darwin/lib/libhashbrown-64c7f86b25c47576.rlib" "/nix/store/j21vazwy4w6c20mhlqpafyxbwj3q226j-rustc-1.50.0/lib/rustlib/x86_64-apple-darwin/lib/librustc_std_workspace_alloc-4d1eb83250258253.rlib" "/nix/store/j21vazwy4w6c20mhlqpafyxbwj3q226j-rustc-1.50.0/lib/rustlib/x86_64-apple-darwin/lib/libunwind-46e0e02619331401.rlib" "/nix/store/j21vazwy4w6c20mhlqpafyxbwj3q226j-rustc-1.50.0/lib/rustlib/x86_64-apple-darwin/lib/libcfg_if-71dbab27ef745144.rlib" "/nix/store/j21vazwy4w6c20mhlqpafyxbwj3q226j-rustc-1.50.0/lib/rustlib/x86_64-apple-darwin/lib/liblibc-8425a559f49d7df9.rlib" "/nix/store/j21vazwy4w6c20mhlqpafyxbwj3q226j-rustc-1.50.0/lib/rustlib/x86_64-apple-darwin/lib/liballoc-3b6462f38288b0eb.rlib" "/nix/store/j21vazwy4w6c20mhlqpafyxbwj3q226j-rustc-1.50.0/lib/rustlib/x86_64-apple-darwin/lib/librustc_std_workspace_core-820a42893aa0c85f.rlib" "/nix/store/j21vazwy4w6c20mhlqpafyxbwj3q226j-rustc-1.50.0/lib/rustlib/x86_64-apple-darwin/lib/libcore-cfb9ec37e32acc9c.rlib" "/nix/store/j21vazwy4w6c20mhlqpafyxbwj3q226j-rustc-1.50.0/lib/rustlib/x86_64-apple-darwin/lib/libcompiler_builtins-36c5611c149fc184.rlib" "-liconv" "-lSystem" "-lresolv" "-lc" "-lm"
  = note: ld: library not found for -liconv
          clang-7: error: linker command failed with exit code 1 (use -v to see invocation)
          

error: aborting due to previous error

error: could not compile `palette`
```

**Comment by jonringer** [comment]
related: #122231

this is probably the more proper change, if the darwin build succeeds, I'll merge

**Comment by jonringer** [comment]
@GrahamcOfBorg build flavours

**Pull request merged**